### ISSUE TITLE
[iOS Simulator] no local address available

### ISSUE THREAD

**Issue by kkalisz** [TM-Apps/konnection#11]
Hello,
First of all thanks for creating this library.
 
I have found issue when testing konnection with iOS simulator issue can be reproduced using **ios-spm** sample  ( tested it with iOS 17.2) I can't test this easily on real device (certificates etc) but there is big change that it works there.

Issue: On iOS simulator "observeNetworkConnection" return ETHERNET connection. Next invocation of "getInfo" returns connection info with empty ipv4 and ipv6 addresses

Cause: I deeply check the issue it looks that on iOS simulator ETHERNET connection is done using "en0" interface. But code related to providing interfaces for given typ of connection "NetworkConnection.networkInterfaces" contains hardcoded values that may not be correct for simulators, but probably is correct for real devices. 

```
    private val NetworkConnection.networkInterfaces: Set<String>?
        get() = when (this) {
            NetworkConnection.WIFI -> setOf("en0")
            NetworkConnection.MOBILE -> setOf("pdp_ip0", "pdp_ip1", "pdp_ip2", "pdp_ip3")
            NetworkConnection.ETHERNET -> setOf("en2", "en3", "en4") //en1
            NetworkConnection.BLUETOOTH_TETHERING,
            NetworkConnection.UNKNOWN_CONNECTION_TYPE -> null
        } 
```

![image](https://github.com/user-attachments/assets/524e822d-2b43-4474-9302-cea7d95f6886)

Possible solutions:
1.  add internal check if device is simulator and add "en0" to allowed ethernet interfaces
2. define fallback interfaces that will be used if base interfaces will not return ipv4 or ipv6.

`IfaddrsInteractor.get(netInterfaces: Set<String>, fallbackInetInterfaces: Set<String> )`

I would be happy to work on this fix, but just let me know if some of proposed solutions is more preferable or maybe you have other ideas.

Best Regards,
Kamil

**Comment by magnumrocha** [comment]
Hi @kkalisz , thank you for your feedback.

In this case, the Simulator gets the connection info from the host system (Mac OS).

So first of all, can you confirm your Mac is connected by the Ethernet?
(just to check if the connection is reflecting the real one)

I like your fix suggestions.
But let's do this check and then discuss the best strategy.

**Comment by kkalisz** [comment]
Hey, 
Indeed connection and connection type is reflection of host machine, in my case:

mac studio:
     en0 : ethernet
     en1 :  wifi 

macbook: (no ethernet port)
     en0 : wifi

So this complicates a bit, because it depends on the host device. 
From obvious reason my 1-st idea will not work, or maybe we can modify it a bit check all "enX" interfaces when device is simulator. 
Personally I would not like to overcomplicate the solution as it looks this only affects testing on simulator.

**Comment by magnumrocha** [comment]
Hi @kkalisz,

I've just released version [1.4.3](https://github.com/TM-Apps/konnection/releases/tag/1.4.3) with the fix for this Issue.

I did some tests with it (on a Simulator and on a real device) and everything has run fine.

Can you please test it as soon as you can?
I'm curious to see your results.

**Comment by kkalisz** [comment]
@magnumrocha thank you for the fix, it workd fine now <3
